### ISSUE TITLE
Barebones web backend - MVP

### ISSUE THREAD

**Issue by ejhumphrey** [omec/bedevere#10]
Blocks: #7, #8, #9
- [x] Barebones flask app on app engine
- [x] `audio/upload` POST endpoint
- [x] `annotation/submit` POST endpoint
- [x] `annotation/taxonomy` GET endpoint

**Comment by stefan-balke** [comment]
Just to make the technical context clear:
https://en.wikipedia.org/wiki/Google_App_Engine

Some basic settings here:
https://github.com/GoogleCloudPlatform/appengine-flask-skeleton

Is this what you mean?

Some **random** questions:
* If so, are using some kind of `requirements.txt` or git submodules for the python dependencies?
* What about the javascript dependencies, `bower` or `npm`?
* Do we already have a preview server?
* Do we want to use bootstrap?
* MongoDB?
* Can we agree on a basic folder architecture first?

**Comment by stefan-balke** [comment]
> Can we agree on a basic folder architecture first?

--> web_backend
--->---> assets
--->--->---> css
--->--->---> js
--->--->---> img
--->---> templates
--->---> requirements/setup
--->--->---> requirements_dev.txt
--->--->---> requirements_production.txt

Something like this?

> What about the javascript dependencies, bower or npm?

Actually I don't like the way the js-packages are installed...maybe it's just easier to put those into the repo directly. Another reason is that most packages contain js and css stuff...but maybe someone has a different view on this.

**Comment by ejhumphrey** [comment]
for more context: https://docs.google.com/document/d/1_Ta7I8liVUq29GVbmvHYhv4iXc_ybw72_POA5GPUdNc/edit?usp=sharing

to the questions:
- yea, we'll use something for dependency management, probably specific to the language things are written in
- I don't think JS deps will touch this issue; they will definitely impact #8 tho
- what do you mean by a preview server? an instance that we can use? I'm going to fire up a free app engine instance tonight, if that's what you mean.
- Bootstrap: maybe? this is another topic for #8
- I'm advocating strongly for using Good Compute Platform (App engine, datastore, and cloud storage) because (a) we'll get very far with what [we can use for free](https://cloud.google.com/appengine/#app-engine-pricing) and (b) we have industry involvement in the project that could _probably_ help us cover any costs.
- folder architecture: this is a great point, and I've opened a new issue (#11) for this.

**Comment by ejhumphrey** [comment]
@stefan-balke hierarchy looks like a good start, can I trouble you to migrate to #11?

**Comment by stefan-balke** [comment]
> an instance that we can use? I'm going to fire up a free app engine instance tonight, if that's what you mean.

Exactly. 👍

**Comment by ejhumphrey** [comment]
for even more context, I'm envisioning the backend server to use python / flask running on a google app engine instance. The tools are shiny, super easy to build with, and (as I understand it) should be nice and scalable.

With how I'm envisioning this specific issue, this should be all of an hour of work.

**Comment by dmcennis** [comment]
Set of requirements for discussion / debate
- [ ] audio
   - [ ]  Submission of audio data
   - [ ] process for validating licenses on content
   - [ ] protocol for requesting content (registeration requirements or not?)
- [ ] Archive of annotations
   - [ ] archive of annotation types
   - [ ] metadata on each annotation type 
   - [ ] content of each annotation
   - [ ] metadata on submitter / confidence / etc. on each annotation
- [ ] metadata on annotation structure
   - [ ] templates for annotation presentation
   - [ ] explanation of the data structures / content in the data format
- [ ] database of registered annotators
   - [ ] Google authentication mechanisms
   - [ ] information on quality and quantity of submissions (database links?)
   - [ ] optional biographical data
- [ ] database of results
   - [ ] content of each submission
   - [ ] metadata on the submission
   - [ ] competition type
   - [ ] references to external algorithm explanations
   - [ ] metadata on the competition as a whole (top / score distribution / evaluation metrics)

What am I missing?
